lib/concordance_validation/delta.rb

```diff
@@ -6,58 +6,40 @@
 module ConcordanceValidation
   # Performs a diff of two validated concordances
   class Delta
-    attr_accessor :old_conc, :new_conc, :adds, :deletes
-
-    def initialize(old_conc_filename, new_conc_filename)
-      @old_conc = open_concordance(old_conc_filename)
-      @new_conc = open_concordance(new_conc_filename)
-      @adds = Hash.new { |h, resolved| h[resolved] = Set.new }
-      @deletes = Hash.new { |h, resolved| h[resolved] = Set.new }
+    def initialize(old_concordance, new_concordance)
+      @old_concordance = old_concordance
+      @new_concordance = new_concordance
     end
 
     def run
-      old_conc_lines = old_conc.readlines.to_set
-      new_conc.each do |line|
-        if old_conc_lines.include? line
-          old_conc_lines.delete(line)
-        else
-          variant_ocn, canonical_ocn = line.chomp.split("\t")
-          adds[canonical_ocn] << variant_ocn
-        end
-      end
-      write(File.open(diff_out_path + ".adds", "w"), adds)
-      deletes_from_remaining_lines(old_conc_lines)
-      write(File.open(diff_out_path + ".deletes", "w"), deletes)
-    end
-
-    def write(fout, diffs)
-      diffs.keys.sort.each do |canonical_ocn|
-        diffs[canonical_ocn].each do |variant_ocn|
-          fout.puts [variant_ocn, canonical_ocn].join("\t")
-        end
-      end
-      fout.flush
+      # Lines only in old concordance == deletes
+      comm_cmd = "bash -c 'comm -23 <(#{sort_cmd @old_concordance}) <(#{sort_cmd @new_concordance}) > #{deletes_file}'"
```

Comment:
I think you can skip the `bash -c` part, and just have `comm_cmd` be:

```ruby
comm_cmd = "comm -23 <(#{sort_cmd @old_concordance}) <(#{sort_cmd @new_concordance}) > #{deletes_file}"
```

Reply:
I think the command needs to specify `bash`, otherwise tests complain ```Syntax error: "(" unexpected```. I think some part of how we're doing `<(subcommand)` is a feature of recent bashes. That was the impression I got.